### DISCUSSION TITLE
MSVC 使用 Module 和标准库模块时无法使用google test

### DISCUSSION THREAD

**Discussion by xiaguan** [xmake-io/xmake#3259] [Q&A]
[我自己的简单项目](https://github.com/xiaguan/SuKV) 
我自己使用vs preview 2022 进行开发，今天我在vs上跑通了gtest，在迁移到xmake里的时候，突然出了很多奇怪的报错。
可能是gtest 和 高标准库冲突了?不是太懂。
xmake.lua 如下
![image](https://user-images.githubusercontent.com/26018364/211147175-334c815f-f43e-43a1-9e2d-b666584e3bb6.png)
报错信息如下
![image](https://user-images.githubusercontent.com/26018364/211147246-ca922909-1423-4d98-90ba-fb867764b506.png)
还有一个问题就是，有时候编译不了，把.xmake目录删除了重新编译就过了，这次不太行。

**Comment by waruqi** [comment]
先试下 dev https://github.com/xmake-io/xmake/issues/3243

xmake update -s dev

**Comment by xiaguan** [comment]
试了一下，又清除了缓存之后，还是一样的问题。
我重新描述一下吧，之前xmake是能用的在我构建SkipList模块的时候，包括import std 。 test.cpp在不引入gtest的时候也能用。模块和标准库模块特性应该是没有问题的。
但今天我先是在vs上加了gtest的测试，确定它能跑以后我再向xmake里的test.cpp迁移的时候，它一直报标准库里有东西重定义。可能不是xmake的问题?可能是gtest的问题我感觉。
